Added method flatten()

islas27 opened the pull request:
- Added new method flatten() to Array.prototype
- Added the unit test
- Added the documentation & examples
- Added new test data

Closes #15

ryanez commented on test/flatten.spec.js:
```diff
@@ -0,0 +1,22 @@
+  })
+
+  it('returns the correct answer', () => {
+    expect(data.unflattenedArrayNumbersS.flatten()).to.eql(data.flattenedArrayNumbersS)
```
Please add a test to validate your method is returning a new array and it's not modifying the original one.

islas27 commented:
Added the unit test and fixed a bit the code, its true it was modifying the original array. Used `slice()` to get a copy and then modify that one with `splice()`. Check commit 0ede140

ryanez commented on aee.js:
```diff
@@ -141,3 +141,15 @@ Array.prototype.min = function (comparer) {
+      copy.splice(i, 1, ...copy[i])
+    }
+  }
+  return copy
```
If you take a look at the whole file this `copy` guy is the black in the rice, I think all the `result` boys will bully him.

islas27 replied:
yeah, it seems it was too late for me to keep working... hahaha, I'm on it!

ryanez commented:
LGTM 👍
